src/app/user/store/user.reducer.ts

```diff
@@ -0,0 +1,22 @@
+import { Action, createReducer, on } from '@ngrx/store';
+
+import { UserActions } from './user.actions';
+
+export const userFeatureKey = 'user';
+
+export interface UserState {
+  isAdmin: boolean;
+  name: string;
+}
+
+export const initialState: UserState = {
+  isAdmin: false,
+  name: '',
+};
+
+const reducer = createReducer(
+  initialState,
+  on(UserActions.requestCurrentUserSuccess, (state, { isAdmin, name }): UserState => ({ ...state, isAdmin, name }))
```

Comment:
Is it necessary to use  .`..state` when all our state consist of two field `isAdmin` and `name`?

Reply:
fixed